diff --git a/ultramodern/src/threads.cpp b/ultramodern/src/threads.cpp
index 6de401d..30c6688 100644
--- a/ultramodern/src/threads.cpp
+++ b/ultramodern/src/threads.cpp
@@ -96,7 +96,7 @@ void ultramodern::set_native_thread_priority(ThreadPriority pri) {
     }
     // SetThreadPriority(GetCurrentThread(), nPriority);
 }
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__ANDROID__)
 #include <sys/prctl.h>
 
 void ultramodern::set_native_thread_name(const std::string& name) {
